ui/client/components/graph/NodeUtils.js

```diff
@@ -205,9 +205,15 @@ class NodeUtils {
   }
 
   //TODO: methods below should be based on backend data, e.g. Subprocess can have outputs or not - based on individual subprocess...
-  hasInputs = (node) => (node.type !== "Source" && node.type !== "SubprocessInputDefinition")
+  hasInputs = (node) => {
```

Comment:
These methods are very similar and could be merged

Reply:
I did a small refactor